cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceTaskContext.java

```diff
@@ -85,6 +86,13 @@
   WorkflowToken getWorkflowToken();
 
   /**
+   * @return information about the enclosing {@link Workflow} run, if this {@link MapReduce} program is executed
+   * as a part of the Workflow; returns {@code null} otherwise. .
+   */
+  @Nullable
+  WorkflowInfo getWorkflowInfo();
```

Comment:
I was thinking about that, but I did not do it, because the Workflow token can by manipulated by the program, whereas the name WorkflowInfo suggests that it is a read-only class. I think for now it is ok to keep them separate.